test/fw/bin/pbs_generate_report

```diff
@@ -0,0 +1,183 @@
+# and the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# OpenPBS is free software. You can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version.
+#
+# OpenPBS is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public
+# License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# PBS Pro is commercially licensed software that shares a common core with
+# the OpenPBS software.  For a copy of the commercial license terms and
+# conditions, go to: (http://www.pbspro.com/agreement.html) or contact the
+# Altair Legal Department.
+#
+# Altair's dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of OpenPBS and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair's trademarks, including but not limited to "PBS™",
+# "OpenPBS®", "PBS Professional®", and "PBS Pro™" and Altair's logos is
+# subject to Altair's trademark licensing policies.
+
+import os
+import csv
+import json
+import sys
+
+def usage():
+    msg = []
+    msg += ['Usage: ' + os.path.basename(sys.argv[0])]
+    msg += [' [benchmark_json_file] [tocompare_json_file] [OPTION]\n\n']
+    msg += [' Performance test report generation tool to generate csv and html report\n\n']
+    msg += ['--html-report : option to generate html report\n']
+    print(''.join(msg))
+
+def generate_html_report():
+    """
+    Generate html performance comparision report
+    """
+    HTML = '''<html>
+    <head>
+      <style>
+        table {
+          font-family: sans-serif, "Times New Roman", serif;
+          border-collapse: collapse;
+          width: 100%%;
+        }
+        td, th {
+          border: 1px solid #dddddd;
+          text-align: left;
+          padding: 8px;
+        }
+      </style>
+    </head>
+    <body>
+      <table>
+        <tr><th><b>Performance tests benchmark comparision results</b></th></tr>
+        <tr><td><b>cmd:</b> %s</td></tr>
+        <tr><td><b>user:</b> %s</td></tr>
+        <tr><td><b>host:</b> %s</td></tr>
+      </table>
+      <table>
+      %s
+      </table>
+    </body>
+    </html>
+    '''
+    with open('performance_test_report.html', 'w+') as fp:
+        fp.write(HTML % (oldv['command'], oldv['user'], list(oldv['machine_info'].keys())[0],
+                         _h + ''.join(_data)))
+
+def generate_csv_report():
+    with open('performance_test_report.csv', 'w+') as fp:
```

Comment:
Please add function headers to all methods.
Also rather than hardcoding file name and creating a file in the current directory, let the user provide path where he/she want to store the file. You can put the default file name in the current directory.